app/v1/controllers/requestParsers/validators/CreateBFLossValidator.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers.validators
+
+import v1.controllers.requestParsers.validators.validations._
+import v1.models.domain.BFLoss
+import v1.models.errors.{MtdError, RuleIncorrectOrEmptyBodyError, RuleInvalidLossAmount, RuleTaxYearNotSupportedError}
+import v1.models.requestData.CreateBFLossRawData
+
+class CreateBFLossValidator extends Validator[CreateBFLossRawData] {
+
+  private val validationSet = List(parameterFormatValidation, bodyFormatValidator, bodyFieldsValidator)
+
+  private def parameterFormatValidation: CreateBFLossRawData => List[List[MtdError]] = { data =>
+    List(
+      NinoValidation.validate(data.nino)
+    )
+  }
+
+  private def bodyFormatValidator: CreateBFLossRawData => List[List[MtdError]] = { data =>
+    List(
+      JsonFormatValidation.validate[BFLoss](data.body.json, RuleIncorrectOrEmptyBodyError)
+    )
+  }
+
+  private def bodyFieldsValidator: CreateBFLossRawData => List[List[MtdError]] = { data =>
+    val req = data.body.json.as[BFLoss]
+    List(
+      TaxYearValidation.validate(req.taxYear),
```

Comment:
If you enter the taxYear field as `"taxYear": "2016"`, you get both `FORMAT_TAX_YEAR` and `RULE_TAX_YEAR_NOT_SUPPORTED`. Are we okay with this, or should we just show one or the other?

Reply:
Fixed in latest update